### DISCUSSION TITLE
Choice of compiler warnings enabled for -Werror

### DISCUSSION THREAD

**Discussion by NathanielMcVicar** [microsoft/onnxruntime#8693] [Other Q&A]
Hi all. I was wondering if there were any motivations behind which warnings are enabled in the onnxruntime build. In our setup (mostly inheriting warnings from LLVM) on Linux I had to remove the following warnings to build onnxruntime without error: cast-qual, covered-switch-default, implicit-fallthrough, non-virtual-dtor, string-conversion, suggest-override. While some of these are matters of opinion, some are pretty fundamental/good practice. Thanks!

**Comment by snnn** [comment]
Welcome to contribute your fixes.
